### PR TITLE
[MSHADE-321] - Always respect 'createDependencyReducedPom' flag

### PR DESCRIPTION
**Pull request by head-thrash**
Always respect 'createDependencyReducedPom' flag.

This PR enables shade-plugin to create DRP disregarding if the final artifact was renamed, but merely based on flag settings.

This PR makes shade plugin behavior aligned with documentation.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSHADE) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHADE-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHADE-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by head-thrash** [comment]
@eolivelli added IT test

**Comment by eolivelli** [comment]
Let's wait for CI
https://builds.apache.org/job/maven-box/job/maven-shade-plugin/job/MSHADE-321/

File: src/it/MSHADE-321_respectDrpFlag/repo/org/apache/maven/its/shade/drp/a/0.1/a-0.1.pom [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+<project>
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.apache.maven.its.shade.drp</groupId>
+  <artifactId>a</artifactId>
+  <version>0.1</version>
+  <packaging>jar</packaging>
+
+  <distributionManagement>
```
Comment (rfscholte) [inline src/it/MSHADE-321_respectDrpFlag/repo/org/apache/maven/its/shade/drp/a/0.1/a-0.1.pom]: Same here, do we need the repository?

**Comment by head-thrash** [comment]
up to maintainers to decide, but this fix actually is applicable and makes plugin behavior more predictable.

**Comment by bakomchik** [comment]
Hi folks) i really want this feture. Could you please merge it and release)

**Comment by michael-o** [comment]
I am willing to merge when @head-thrash can answer the questions. If there won't be any objections, I will merge somewhere next week. @bakomchik feel free to ping me.

**Comment by head-thrash** [comment]
> Please answer Robert's questions.

@michael-o answered. odd dependencyManagement removed. hope this unblocks this issue.

**Comment by jira-importer** [comment]
Resolve #321

**Pull request closed without merging**